code/components/extra-natives-five/src/RuntimeAssetNatives.cpp

```diff
@@ -1231,8 +1214,8 @@ static InitFunction initFunction([]()
 					fwEntityDef* entityDef = (fwEntityDef*)MakeStructFromMsgPack("CEntityDef", entityData);
 					mapData->entities.Set(i, entityDef);
 
-					uint64_t archetypeUnk = 0xFFFFFFF;
-					fwArchetype* archetype = GetArchetypeSafe(entityDef->archetypeName, &archetypeUnk);
+					rage::fwModelId modelId{ 0xFFFFFFF };
```

Comment:
Some more context on this `0xFFFFFFF` (7 Fs, magic number) and others down the line is necessary. Some ideas: a constant (in) or a static method in `rage::fwModelId` could also be an option.

Reply:
`fwModelId` probably is a bitfield (28 bit handle, 4 bits for flags/type) and the handle being `-1` feels like a default for 'invalid'.

This could likely be a constructor, either default or with a magic type (like how nullptr_t/nullopt_t work).

Reply:
Checking game code (fwMapData::ConstructEntity in companion), it appears the lower 16 bits are the model index, the 12 bits above are the index of a map types file, and that leaves 4 bits of flags.